### PR TITLE
Use karma concurrency for initiating parallel sauce tests

### PR DESCRIPTION
**Pull request by megawac**
As of 0.13.12 karma supports a concurrency field to limit browsers started in parallel. See karma-runner/karma#1646

Don't merge til green lights

Have to still tinker with the concurrency limit. I think we should be able to set 10

Webkit based browsers will currently pass all tests but give an error. See karma-runner/karma#1101 & karma-runner/karma#1648

### CODE REVIEW

**Comment by megawac** [comment]
@jdalton I notice something odd when testing microsoftedge browser locally. Any thoughts?

```sh
$ karma start karma.conf-sauce.js 
INFO [karma]: Karma v0.12.31 server started at http://localhost:9876/
INFO [launcher]: Starting browser microsoftedge 20 (Windows 10) on SauceLabs
INFO [launcher.sauce]: microsoftedge 20 (Windows 10) session at https://saucelabs.com/tests/d9a4434352c1487ab7de5252e1713268
INFO [Chrome 42.0.2311 (Windows)]: Connected on socket UUMualhhWIhYuLOu-Nb1 with id 57579994
................................................................................
................................................................................
................................................................................
................................................................................
........................................................................
Chrome 42.0.2311 (Windows): Executed 392 of 392 SUCCESS (1.416 secs / 1.383 secs)
INFO [launcher.sauce]: Shutting down Sauce Connect
```

Notice it says Chrome 42.0.2311 on complete and microsoft edge on start

**Comment by jdalton** [comment]
Looks like a UA sniff issue on Karma's complete parts.

**Comment by megawac** [comment]
Chrome issues are a result of https://github.com/karma-runner/karma/issues/1101 & https://github.com/karma-runner/karma/pull/1648

File: karma.conf-sauce.js [hunk 1]
```diff
@@ -26,7 +26,10 @@ var sauceBrowsers = _.reduce([
 
   ['android', '5'],
   ['android', '4.4'],
-  ['android', '4.3'],
+
+  // 4.3 currently erros with some router tests
+  // ['android', '4.3'],
```
Comment (megawac) [inline karma.conf-sauce.js]: By the way, heres the Android 4.3 browser error

```sh
[32m23 10 2015 13:26:40.075:INFO [karma]: [39mKarma v0.13.13 server started at http://localhost:9876/
[32m23 10 2015 13:26:40.086:INFO [launcher]: [39mStarting browser android 4.3 on SauceLabs
[32m23 10 2015 13:27:33.141:INFO [launcher.sauce]: [39mandroid 4.3 session at https://saucelabs.com/tests/db7f9739617f49e88b77f07d2654e652
[32m23 10 2015 13:27:34.618:INFO [Android 4.3.0 (Android 4.3.0)]: [39mConnected on socket aoXmsl3pRitgwaw2AAAA with id 54729711
................................................................................
................................................................................
................................................................................
....................................
Android 4.3.0 (Android 4.3.0) Backbone.Router routes (simple) FAILED
	    at Object.<anonymous> (/home/gyeates/code/backbone/test/router.js:189:12)
	    at Object.run (/home/gyeates/code/backbone/node_modules/qunitjs/qunit/qunit.js:810:28)
	    at /home/gyeates/code/backbone/node_modules/qunitjs/qunit/qunit.js:942:11
	    at process (/home/gyeates/code/backbone/node_modules/qunitjs/qunit/qunit.js:624:24)
	    at begin (/home/gyeates/code/backbone/node_modules/qunitjs/qunit/qunit.js:606:2)
	    at /home/gyeates/code/backbone/node_modules/qunitjs/qunit/qunit.js:666:4
	
Android 4.3.0 (Android 4.3.0) Backbone.Router routes (simple) FAILED
	    at Object.<anonymous> (/home/gyeates/code/backbone/test/router.js:189:12)
	    at Object.run (/home/gyeates/code/backbone/node_modules/qunitjs/qunit/qunit.js:810:28)
	    at /home/gyeates/code/backbone/node_modules/qunitjs/qunit/qunit.js:942:11
	    at process (/home/gyeates/code/backbone/node_modules/qunitjs/qunit/qunit.js:624:24)
	    at begin (/home/gyeates/code/backbone/node_modules/qunitjs/qunit/qunit.js:606:2)
	    at /home/gyeates/code/backbone/node_modules/qunitjs/qunit/qunit.js:666:4
	
Android 4.3.0 (Android 4.3.0) Backbone.Router routes (simple, but unicode) FAILED
	    at Object.<anonymous> (/home/gyeates/code/backbone/test/router.js:199:12)
	    at Object.run (/home/gyeates/code/backbone/node_modules/qunitjs/qunit/qunit.js:810:28)
	    at /home/gyeates/code/backbone/node_modules/qunitjs/qunit/qunit.js:942:11
	    at process (/home/gyeates/code/backbone/node_modules/qunitjs/qunit/qunit.js:624:24)
	    at begin (/home/gyeates/code/backbone/node_modules/qunitjs/qunit/qunit.js:606:2)
	    at /home/gyeates/code/backbone/node_modules/qunitjs/qunit/qunit.js:666:4
	
Android 4.3.0 (Android 4.3.0) Backbone.Router routes (simple, but unicode) FAILED
	    at Object.<anonymous> (/home/gyeates/code/backbone/test/router.js:199:12)
	    at Object.run (/home/gyeates/code/backbone/node_modules/qunitjs/qunit/qunit.js:810:28)
	    at /home/gyeates/code/backbone/node_modules/qunitjs/qunit/qunit.js:942:11
	    at process (/home/gyeates/code/backbone/node_modules/qunitjs/qunit/qunit.js:624:24)
	    at begin (/home/gyeates/code/backbone/node_modules/qunitjs/qunit/qunit.js:606:2)
	    at /home/gyeates/code/backbone/node_modules/qunitjs/qunit/qunit.js:666:4
	
.....................
Android 4.3.0 (Android 4.3.0) Backbone.Router correctly handles URLs with % (#868) FAILED
	    at Object.<anonymous> (/home/gyeates/code/backbone/test/router.js:417:12)
	    at Object.run (/home/gyeates/code/backbone/node_modules/qunitjs/qunit/qunit.js:810:28)
	    at /home/gyeates/code/backbone/node_modules/qunitjs/qunit/qunit.js:942:11
	    at process (/home/gyeates/code/backbone/node_modules/qunitjs/qunit/qunit.js:624:24)
	    at begin (/home/gyeates/code/backbone/node_modules/qunitjs/qunit/qunit.js:606:2)
	    at /home/gyeates/code/backbone/node_modules/qunitjs/qunit/qunit.js:666:4
	
Android 4.3.0 (Android 4.3.0) Backbone.Router correctly handles URLs with % (#868) FAILED
	    at Object.<anonymous> (/home/gyeates/code/backbone/test/router.js:417:12)
	    at Object.run (/home/gyeates/code/backbone/node_modules/qunitjs/qunit/qunit.js:810:28)
	    at /home/gyeates/code/backbone/node_modules/qunitjs/qunit/qunit.js:942:11
	    at process (/home/gyeates/code/backbone/node_modules/qunitjs/qunit/qunit.js:624:24)
	    at begin (/home/gyeates/code/backbone/node_modules/qunitjs/qunit/qunit.js:606:2)
	    at /home/gyeates/code/backbone/node_modules/qunitjs/qunit/qunit.js:666:4
	
..................................................
Android 4.3.0 (Android 4.3.0) Backbone.sync destroy FAILED
	Expected: null
	Actual: undefined
	    at Object.<anonymous> (/home/gyeates/code/backbone/test/sync.js:127:12)
	    at Object.run (/home/gyeates/code/backbone/node_modules/qunitjs/qunit/qunit.js:810:28)
	    at /home/gyeates/code/backbone/node_modules/qunitjs/qunit/qunit.js:942:11
	    at process (/home/gyeates/code/backbone/node_modules/qunitjs/qunit/qunit.js:624:24)
	    at begin (/home/gyeates/code/backbone/node_modules/qunitjs/qunit/qunit.js:606:2)
	    at /home/gyeates/code/backbone/node_modules/qunitjs/qunit/qunit.js:666:4
	
Android 4.3.0 (Android 4.3.0) Backbone.sync destroy FAILED
	Expected: null
	Actual: undefined
	    at Object.<anonymous> (/home/gyeates/code/backbone/test/sync.js:127:12)
	    at Object.run (/home/gyeates/code/backbone/node_modules/qunitjs/qunit/qunit.js:810:28)
	    at /home/gyeates/code/backbone/node_modules/qunitjs/qunit/qunit.js:942:11
	    at process (/home/gyeates/code/backbone/node_modules/qunitjs/qunit/qunit.js:624:24)
	    at begin (/home/gyeates/code/backbone/node_modules/qunitjs/qunit/qunit.js:606:2)
	    at /home/gyeates/code/backbone/node_modules/qunitjs/qunit/qunit.js:666:4
	
.........................................
Android 4.3.0 (Android 4.3.0): Executed 392 of 392 (4 FAILED) (1.207 secs / 0.905 secs)
[32m23 10 2015 13:27:37.141:INFO [launcher.sauce]: [39mShutting down Sauce Connect

```

/cc @jridgewell

Comment (jridgewell) [inline karma.conf-sauce.js]: > /cc @jridgewell

Hm? @braddunbar would have better luck with a url issue.

**Comment by megawac** [comment]
This should be good to go at any rate. Still requires a look into the Android 4 issues /cc @braddunbar

**Pull request merged**